stage1/init/common/pod.go

```diff
@@ -256,24 +259,44 @@ func appToSystemd(p *stage1commontypes.Pod, ra *schema.RuntimeApp, interactive b
 		return errwrap.Wrap(errors.New("unable to write environment file"), err)
 	}
 
-	// This is a partial implementation for app.User and app.Group:
-	// For now, only numeric ids (and the string "root") are supported.
 	var uid, gid int
 	var err error
-	if app.User == "root" {
-		uid = 0
+
+	if strings.HasPrefix(app.User, "/") {
+		var stat syscall.Stat_t
+		if err = syscall.Lstat(strings.Join([]string{common.AppRootfsPath(p.Root, appName),
+			app.User}, "/"), &stat); err != nil {
+			return errwrap.Wrap(fmt.Errorf("unable to get uid from file %q",
+				app.User), err)
+		}
+		uid = int(stat.Uid)
```

Comment:
I think you might have misread that line. The "/" is the separator argument for `strings.Join` which isn't appended.